runtime/src/tasks/qthreads/tasks-qthreads.c

```diff
@@ -913,36 +913,32 @@ chpl_taskID_t chpl_task_getId(void)
 
 void chpl_task_sleep(double secs)
 {
-    if (qthread_shep() == NO_SHEPHERD) {
-        struct timeval deadline;
-        struct timeval now;
-
-        //
-        // Figure out when this task can proceed again, and until then, keep
-        // yielding.
-        //
-        gettimeofday(&deadline, NULL);
-        deadline.tv_usec += (suseconds_t) ((secs - trunc(secs)) * 1.0e6);
-        if (deadline.tv_usec > 1000000) {
-            deadline.tv_sec++;
-            deadline.tv_usec -= 1000000;
-        }
-        deadline.tv_sec += (time_t) trunc(secs);
+    qthread_shepherd_id_t shep = qthread_shep();
+    struct timeval deadline;
+    struct timeval now;
+
+    //
+    // Figure out when this task can proceed again, and until then, keep
+    // yielding.
+    //
+    gettimeofday(&deadline, NULL);
+    deadline.tv_usec += (suseconds_t) lround((secs - trunc(secs)) * 1.0e6);
+    if (deadline.tv_usec > 1000000) {
+        deadline.tv_sec++;
+        deadline.tv_usec -= 1000000;
+    }
+    deadline.tv_sec += (time_t) trunc(secs);
 
-        do {
+    do {
+        if (shep == NO_SHEPHERD) {
```

Comment:
I thought it worthwhile from a wakeup-latency point of view to take the if-stmt on the shepherd value out of the spin loop.  But that's probably moot since I'm strongly considering moving the whole of sleep() into the module code and getting rid of the runtime interface entirely.  (Though I'd like to complete this change first so as to give myself scheduling leeway on that latter one.)